vault/logical_system.go

```diff
@@ -1869,6 +1899,47 @@ func (b *SystemBackend) handleAuthTable(ctx context.Context, req *logical.Reques
 	return resp, nil
 }
 
+func expandStringValsWithCommas(configMap map[string]interface{}) error {
+	configParamNameSlice := []string{
+		"audit_non_hmac_request_keys",
+		"audit_non_hmac_response_keys",
+		"passthrough_request_headers",
+		"allowed_response_headers",
+	}
+	for _, paramName := range configParamNameSlice {
+		if raw, ok := configMap[paramName]; ok {
+			outputSlice := []string{}
+
+			switch t := raw.(type) {
+			case []interface{}:
```

Comment:
So maybe we do nothing in the slice case, for compatibility reasons.  Rationale: we only expand the commas as a convenience for users, because sometimes it's easier to produce a single comma-separated value than an array; if they can already produce an array, they don't need that convenience.  As a side benefit, it may provide a workaround if for some reason they need a comma in one these inputs: just use an array with an empty element.  ("may" because presumably not all endpoints tolerate an empty array element gracefully.)

Reply:
To clarify, if they pass in ["abc,def", "ghi"], then we want to preserve the comma in the first element, right? 
Then, both auth enable and auth tune would work the same.

Reply:
Yes.

Reply:
Do we need to keep the []interface{} case?  It doesn't seem like it does anything now.

Reply:
not really! Done